consensus/parlia/parlia.go

```diff
@@ -381,11 +388,15 @@ func getValidatorBytesFromHeader(header *types.Header, chainConfig *params.Chain
 		return nil
 	}
 	num := int(header.Extra[extraVanity])
-	if num == 0 || len(header.Extra) <= extraVanity+extraSeal+num*validatorBytesLength {
-		return nil
-	}
 	start := extraVanity + validatorNumberSize
 	end := start + num*validatorBytesLength
+	extraMinLen := end + extraSeal
```

Comment:
The extra dynamic part is so complex now, could we use rlp to encode the custom data into extra?

```golang
|---Extra Vanity---|---RLP(CustomData)---|---Extra Seal---|
|---Fixed 32 bytes---|---dynamic length---|---Fixed 65 bytes---|
```

Reply:
> The extra dynamic part is so complex now, could we use rlp to encode the custom data into extra?
> 
> ```go
> |---Extra Vanity---|---RLP(CustomData)---|---Extra Seal---|
> |---Fixed 32 bytes---|---dynamic length---|---Fixed 65 bytes---|
> ```

good idea!
I think we can do this optimization in another seperate PR